utils/bundle_miner.c

```diff
@@ -61,7 +62,7 @@ static void *bundle_miner_mine_routine(void *const param) {
       if (probability < ctx->probability) {
         ctx->probability = probability;
         ctx->optimal_index = ctx->index;
-        if ((uint32_t)(1 / probability) >= ctx->mining_threshold) {
+        if (num_trials_mining_threshold > 1 && (uint64_t)(1 / probability) >= num_trials_mining_threshold) {
```

Comment:
I think `(1 / probability)` would first be implicitly promoted to something smaller than `uint64_t` then explicitly casted to `uint64_t `; hence, losing some of its value. Just to be sure I think we should suffix `1` with `ULL`.

Reply:
great

Reply:
Actually, i think it should be 1.0